File: factory/TaskHandler.py
from enum import Enum

class TaskActionFactory:
    def create_task_action(self, intent):
        if isinstance(intent, SpotifyIntent):
            return SpotifyTaskAction().handle_action(intent)
        elif isinstance(intent, ChatGPTIntent):
            return ChatGPTTaskAction().handle_action(intent)
        else:
            raise ValueError("Intent Not Found")
        
class SpotifyIntent(Enum):
    ACTION = "Get Albums"

class ChatGPTIntent(Enum):
    ACTION = "Ask Question"

class SpotifyTaskAction:
    def handle_action(self, intent):
        print(intent.value)

class ChatGPTTaskAction:
    def handle_action(self, intent):
        print(intent.value)
    

# Main
factory = TaskActionFactory()
spotify_intent = SpotifyIntent.ACTION
chatgpt_intent = ChatGPTIntent.ACTION

result = factory.create_task_action(chatgpt_intent)

File: factory/TaskHandler.java
interface TaskAction {
    void handleAction();
}

class SpotifyTaskAction implements TaskAction {

    @Override
    public void handleAction() {
        System.out.println("Spotify");
    } 
}

class OpenAITaskAction implements TaskAction {

    @Override
    public void handleAction() {
        System.out.println("OpenAI");
    } 
}

interface TaskActionFactory {
    TaskAction createTaskAction();
}

class SpotifyFactory implements TaskActionFactory {

    @Override
    public TaskAction createTaskAction() {
        return new SpotifyTaskAction();
    }

}

class OpenAIFactory implements TaskActionFactory {

    @Override
    public TaskAction createTaskAction() {
        return new OpenAITaskAction();
    }

}

class Main {
    public static void main(String[] args) {
        TaskActionFactory spotifyFactory = new SpotifyFactory();
        TaskActionFactory openaiFactory = new OpenAIFactory();

        TaskAction spotifyTaskAction = spotifyFactory.createTaskAction();
        TaskAction openaiTaskAction = openaiFactory.createTaskAction();

        spotifyTaskAction.handleAction();
        openaiTaskAction.handleAction();
    }
}
